ysh/expr_to_ast.py

```diff
@@ -1124,17 +1124,17 @@ def _Suite(self, pnode):
 
         return command.CommandList(self.func_items(pnode.GetChild(items_index)))
 
-    def TeaFunc(self, pnode, out):
+    def NamedFunc(self, pnode, out):
         # type: (PNode, command.Func) -> None
         """Parse tree to LST
 
-        tea_func:
-          '(' [func_params] [';' func_params] ')' [type_expr_list] suite 
+        named_func: Expr_Name '(' [func_params] [';' func_params] ')'
         """
-        assert pnode.typ == grammar_nt.tea_func
-        assert pnode.GetChild(0).tok.id == Id.Op_LParen  # proc foo(
+        assert pnode.typ == grammar_nt.named_func
+
```

Comment:
FWIW I added self.p_printer in this file which can help visualize the `pnode`, which follows the grammar directly

Reply:
Thanks for mentioning this! It made cleaning this function up much easier.